metered-channel/src/lib.rs

```diff
@@ -101,15 +107,26 @@ impl Meter {
 	}
 
 	fn note_sent(&self) -> usize {
+		#[cfg(feature = "futures_channel")]
+		self.channel_len.fetch_add(1, Ordering::Relaxed);
 		self.sent.fetch_add(1, Ordering::Relaxed)
 	}
 
 	fn retract_sent(&self) {
 		self.sent.fetch_sub(1, Ordering::Relaxed);
+		#[cfg(feature = "futures_channel")]
+		self.channel_len.fetch_add(1, Ordering::Relaxed);
+	}
+
+	#[cfg(feature = "async_channel")]
+	fn note_channel_len(&self, len: usize) {
```

Comment:
I don't like the idea to do what's already done twice, as async_channel has `len` method.